submissions/AnnaStoyano/Friends-app/index.js

```diff
@@ -0,0 +1,194 @@
+document.addEventListener('DOMContentLoaded', function () {
+    const API_PATH = 'https://randomuser.me/api/';
+    const USER_COUNT = 20;
+    const URL_FOR_SEARCH = `${API_PATH}?results=${USER_COUNT}`;
+
+    fetch(URL_FOR_SEARCH)
+        .then(resolve => resolve.json())
+        .then(data => {
+            const users = data.results;
+            const friendsApp = new FriendsApp(users);
+            friendsApp.start();
+        });
+});
+
+class FriendsApp {
+    constructor(users) {
+        this.resultsWrapper = document.querySelector('.result');
+        this.cards = users.map(item => new UserCard(item));
+        this.dipslayCards = this.cards;
+        this.prevDisplay = this.dipslayCards;
+        this.filterByName();
+        this.filterByAge();
+        this.filterByGender();
+        this.filterByCity();
+        this.filterByEmail();
+    }
+
+    start() {
+        this.render(this.dipslayCards);
+    }
+
+    render() {
+        this.resultsWrapper.innerHTML = '';
+        this.dipslayCards.forEach(card => this.resultsWrapper.insertAdjacentElement('afterbegin', card.card));
+    }
+
+    filterByName() {
+        this.filterByNameAlphabetically();
+        this.filterByNameNonAlphabetically();
+    }
+
+    filterByNameAlphabetically() {
```

Comment:
Confusing naming - this function not about filtering, it is about adding listeners or maybe initialising. Please, change name.

Also, you can add one listener on the wrapper instead of two handlers.